Chummer/Classes/clsCustomDataDirectoryInfo.cs

```diff
@@ -0,0 +1,475 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Windows.Forms;
+using System.Xml;
+
+namespace Chummer
+{
+    public class CustomDataDirectoryInfo : IComparable, IEquatable<CustomDataDirectoryInfo>, IComparable<CustomDataDirectoryInfo>
+    {
+        private bool _blnCreated;
+        private int _version;
+
+        //Tuple<int Hash, string Name, int Version, bool isMinimumVersion>
+        private readonly List<Tuple<Guid, string, int, bool>> _lstDependencies= new List<Tuple<Guid, string, int, bool>>();
+        //Tuple<int Hash, string Name, int Version, bool ignoreVersion>
+        private readonly List<Tuple<Guid, string, int, bool>> _lstExclusivities = new List<Tuple<Guid, string, int, bool>>();
```

Comment:
That's an interesting idea and those tuples are honestly really hard to read.
A struct would certainly be better for readability, I'm certainly gonna take a look at that.

The contained strings and other values are probably bigger than the suggested maximum sizes for structs, but _maybe_ sacrificing some performance for readability and maintainability could worth.

Reply:
I'm a fan of classes just because I often started out with a struct just to change it to a class later, when I had to maintain the codebase and add some logic, checks of constructor-magic to it. 

My "gut"-feeling would be: Just make it a List<MyDependencyObject> and put everything you need in properties.